### ISSUE TITLE
SimCity 2013 - minor rendering issues

### ISSUE THREAD

**Issue by firescry** [doitsujin/dxvk#2361]
With the latest fixes in dxvk, major issue  (#1869) is now resolved and the game is playable. Few other rendering problems became visible though:
- trees are incorrectly rendered, they are replaced by other models (pic. 1, 2)
- some animations look wrong and are played too fast e.g. wind turbine propeller (pic. 1)
- building outline is incorrect (pic. 3)
- some building parts are missing/relocated, e.g. elements at the top of chimneys (pic. 3)
- there is a grid rendered on the ground (pic. 1, 2, 3)

Issues are not visible for wine d3d. Attached apitrace covers all mentioned problems.

| # | wine d3d | dxvk |
| --- | --- | --- |
| 1 | ![0009837814_wine](https://user-images.githubusercontent.com/49809580/141659202-dbd0c896-9285-4ed0-85f5-c0c5138d64e9.jpg) | ![0009837814_dxvk](https://user-images.githubusercontent.com/49809580/141659221-8c877bd4-7e7c-4c62-940c-4eb07621e922.jpg) |
| 2 | ![0014769307_wine](https://user-images.githubusercontent.com/49809580/141659248-d92351da-f193-4136-9afe-fc1a35798fa5.jpg) | ![0014769307_dxvk](https://user-images.githubusercontent.com/49809580/141659261-ce1bf11c-82fd-487e-85bf-315ceab14a7a.jpg) |
| 3 | ![0025901796_wine](https://user-images.githubusercontent.com/49809580/141659293-3d6c85c2-57fa-4fee-ac21-70e21ee17a1e.jpg) | ![0025901796_dxvk](https://user-images.githubusercontent.com/49809580/141659300-b4fb431e-5426-4f48-8aa2-097d6d726133.jpg) |

### Software information
SimCity 2013 (Origin)

### System information
- GPU: AMD Radeon RX 580
- Driver: mesa 21.2.5 @ linux 5.15.2
- Wine version: wine-6.21 (Staging)
- DXVK version: v1.9.2-43-g12249fd3 (12249fd)

### Apitrace file(s)
- [SimCity_1.tar.xz](https://drive.google.com/file/d/1C7YpG83OfrDRlH-IDpomAYH5mEfcZGcI/view?usp=sharing)

### Log files
- d3d9.log: [SimCity_d3d9.log](https://github.com/doitsujin/dxvk/files/7532735/SimCity_d3d9.log)
- d3d11.log: log was empty
- dxgi.log: log was empty

**Comment by firescry** [comment]
Thank you for providing fixes 😊

I checked the patches from #2364 and game looks much better now. The only problem that is still visible is the grid on the ground. It may not be very noticeable on the screenshots above, but this diff shows it better (diff between frame rendered by wine d3d vs. dxvk):
![diff](https://user-images.githubusercontent.com/49809580/143780167-f0556bfc-c566-42d9-8cf4-ac64871ed294.png)

**Comment by K0bin** [comment]
Apitrace of that please.

**Comment by firescry** [comment]
Here is the new apitrace that shows specifically the issue of grid on the ground - [SimCity_2.tar.xz](https://drive.google.com/file/d/1nTFmLyPVHEqepOK3TK4MuV-WcTcyyEDt/view?usp=sharing). The grid is visible also on pic. 1 below. 

Additionally I've noticed that in camera lowering animation (just after map is loaded), there is an extra shadow visible on the left hand side. It looks like shadow cast by big square (pic. 2). This is visible only shortly after map is loaded and disappears quickly. It's not very disturbing, but I'm reporting it here just in case you'll find it interesting to check it further (it's captured in the apitrace above).

| # | wine d3d | dxvk |
|---|---|---|
| 1 | ![wine0046190043](https://user-images.githubusercontent.com/49809580/143785792-f4aea848-ddd7-4e6b-b9bb-421f15031fb0.jpg) | ![dxvk0046190043](https://user-images.githubusercontent.com/49809580/143785804-8fca39a7-3d68-41eb-8f30-4d4dc8223ea0.jpg) |
| 2 | ![wine0003522776](https://user-images.githubusercontent.com/49809580/143785842-e5ceb0a1-0a5a-4924-baae-3e3a1cee3d6a.jpg) | ![dxvk0003522776](https://user-images.githubusercontent.com/49809580/143785851-73d0c6bc-f457-41aa-be9a-6fbf8cfee1bb.jpg) |

**Comment by CME42** [comment]
1. The game is using a cube map with a different texture on each face to render the terrain and it relies on the cube map to be seamful, similar bug like in #2239.

2. I can reproduce it on AMD but not on NV, so it might be a driver issue, but i haven't looked at it any further yet.

**Comment by K0bin** [comment]
FWIW I can reproduce it on Nvidia.

**Comment by K0bin** [comment]
Reopening this because of the grid issue. The problem with trees is fixed though.

**Comment by firescry** [comment]
Thank you for checking on the problem with the grid ☺️

As for the issue with extra shadow (pic. 2), I do not have access to any other GPU, but I tried with AMDVLK instead of RADV and this shadow is still there. I used AMDVLK 2021.Q4.1 (currently the latest) and here is the dxvk log - [SimCity_d3d9.log](https://github.com/doitsujin/dxvk/files/7654175/SimCity_d3d9.log).

**Comment by CME42** [comment]
Yeah sorry i forgot to comment here, the shadow issue happens due to the d24s8 -> d32s8 fixup that is necessary on AMD. I can also repro it on NV when i force the fixup there - so it is also a DXVK bug.

**Comment by K0bin** [comment]
Probably #2892
